BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnostics/SelfDiagnosticsConfigRefresher.cs

```diff
@@ -0,0 +1,120 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.SelfDiagnostics
+{
+    using System;
+    using System.Diagnostics;
+    using System.Diagnostics.Tracing;
+    using System.IO;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// SelfDiagnosticsConfigRefresher class checks a location for a configuration file
+    /// and open a MemoryMappedFile of a configured size at the configured file path.
+    /// The class provides a stream object with proper write position if the configuration
+    /// file is present and valid. Otherwise, the stream object would be unavailable,
+    /// nothing will be logged to any file.
+    /// </summary>
+    internal class SelfDiagnosticsConfigRefresher : IDisposable
+    {
+
+        private const int ConfigurationUpdatePeriodMilliSeconds = 10000;
+
+        private readonly CancellationTokenSource cancellationTokenSource;
+        private readonly Task worker;
+        private readonly SelfDiagnosticsConfigParser configParser;
+        private readonly MemoryMappedFileHandler memoryMappedFileHandler;
+
+
+        private bool disposedValue;
+
+        // Once the configuration file is valid, an eventListener object will be created.
+        // Commented out for now to avoid the "field was never used" compiler error.
+        private SelfDiagnosticsEventListener eventListener;
+
+        private EventLevel logEventLevel = (EventLevel)(-1);
+
+        public SelfDiagnosticsConfigRefresher()
+        {
+            this.configParser = new SelfDiagnosticsConfigParser();
+            this.memoryMappedFileHandler = new MemoryMappedFileHandler();
+            this.UpdateMemoryMappedFileFromConfiguration();
+            this.cancellationTokenSource = new CancellationTokenSource();
+            this.worker = Task.Run(() => this.Worker(this.cancellationTokenSource.Token), this.cancellationTokenSource.Token);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
```

Comment:
Is there a test case to validate to check what happens if `Dispose`  is called twice?

Reply:
Had discussion with Sean, we could ignore this for now.